examples/other_datasets/financial_data.py

```diff
@@ -0,0 +1,186 @@
+
+
+class ToEpochs(TransformerMixin, BaseEstimator):
+    def __init__(self, n):
+        self.n = n
+
+    def fit(self, X, y):
+        return self
+
+    def transform(self, X):
+        all_epochs = []
+        for x in X:
+            id = x[3]
+            epoch = features[features.index > id - self.n]
+            epoch = epoch[epoch.index <= id]
+            epoch.drop(columns=["index"], inplace=True)
+            all_epochs.append(epoch)
+        all_epochs = np.array(all_epochs)
+        return all_epochs
+
+
+def slim(x, keep_diagonal=True):
+    # Vectorize covariance matrices by removing redundant information.
+    l = len(x) // 2
+    first = range(0, l)
+    last = range(len(x) - l, len(x))
+    down_cadrans = x[np.ix_(last, last)]
+    if keep_diagonal:
+        down_cadrans = [down_cadrans[i, j] for i in first for j in first if i <= j]
+    else:
+        down_cadrans = [down_cadrans[i, j] for i in first for j in first if i < j]
+    first_cadrans = np.reshape(x[np.ix_(last, first)], (1, len(x)))
+    ret = np.append(first_cadrans, down_cadrans)
+    return ret
+
+
+class SlimVector(TransformerMixin, BaseEstimator):
+    def __init__(self, keep_diagonal):
+        self.keep_diagonal = keep_diagonal
+
+    def fit(self, X, y):
+        return self
+
+    def transform(self, X):
+        return np.array([slim(x, self.keep_diagonal) for x in X])
+
+
+class OptionalWhitening(TransformerMixin, BaseEstimator):
+    def __init__(self, process=True, n_components=4):
+        self.process = process
+        self.n_components = n_components
+
+    def fit(self, X, y):
+        return self
+
+    def transform(self, X):
+        if not self.process:
+            return X
+        return Whitening(dim_red={"n_components": 4}).fit_transform(X)
+
+
+# Finally put together the transformers, and add at the end
+# the SVM classifier (classical)
+pipe = make_pipeline(
+    ToEpochs(n=10),
+    XdawnCovariances(nfilter=1),
+    OptionalWhitening(process=True, n_components=4),
+    SlimVector(keep_diagonal=True),
+    QuanticSVM(quantum=False),
+)
+
+# Optimize the pipeline.
+# Let's save some time and run the optimization with a classical SVM.
+gs = GridSearchCV(
+    pipe,
+    param_grid={
+        "toepochs__n": [10, 20],
+        "xdawncovariances__nfilter": [1, 2],
+        "optionalwhitening__process": [True, False],
+        "optionalwhitening__n_components": [2, 4],
+        "slimvector__keep_diagonal": [True, False],
+    },
+    scoring="balanced_accuracy",
+)
+
+##############################################################################
+# Run evaluation
+# ----------------
+#
+# Balance the data and run the evaluation on a quantum vs classical pipeline.
+
+# We only have about 200 frauds epochs over 30K entries.
+# Let's balance the problem using NearMiss.
+# Note: at this stage `features` also contains the `index` column.
+# So `NearMiss` we choose the closed 200 non-fraud epochs to the 200 fraud-epochs
```

Comment:
Did you mean [closest](https://www.google.com/search?q=closest+definition&rlz=1C1CHBF_enUS831US831&oq=closest+definition&gs_lcrp=EgZjaHJvbWUyCQgAEEUYORiABDIKCAEQABgPGBYYHjIICAIQABgWGB4yCAgDEAAYFhgeMggIBBAAGBYYHjIICAUQABgWGB4yCAgGEAAYFhgeMggIBxAAGBYYHjIKCAgQABiGAxiKBTIKCAkQABiGAxiKBdIBCDM3ODdqMWo3qAIAsAIA&sourceid=chrome&ie=UTF-8) here instead of [closed](https://www.google.com/search?q=closed+definition&sca_esv=574441619&rlz=1C1CHBF_enUS831US831&sxsrf=AM9HkKnNhlkk4Np83RO8bDVl9EViF0gw8A%3A1697635453454&ei=fdwvZbaSG5zV5NoP4tGagAU&ved=0ahUKEwj2xKTx2P-BAxWcKlkFHeKoBlAQ4dUDCBA&uact=5&oq=closed+definition&gs_lp=Egxnd3Mtd2l6LXNlcnAiEWNsb3NlZCBkZWZpbml0aW9uMgYQABgHGB4yBhAAGAcYHjIGEAAYBxgeMgYQABgHGB4yBhAAGAcYHjIFEAAYgAQyBhAAGAcYHjIGEAAYBxgeMgYQABgHGB4yBhAAGAcYHki1EFCDAlidCHABeAGQAQCYAaEBoAGlA6oBAzAuM7gBA8gBAPgBAcICChAAGEcY1gQYsAPCAgoQABiKBRiwAxhDwgILEAAYBxgeGPEEGArCAg0QABiABBixAxhGGPkBwgIkEAAYgAQYsQMYRhj5ARiXBRiMBRjdBBhGGPQDGPUDGPYD2AEB4gMEGAAgQYgGAZAGCboGBggBEAEYEw&sclient=gws-wiz-serp) here.